### PR TITLE
Entity Behaviors and Entities

### PR DESCRIPTION
**Pull request by Enes5519**
## TODO:

**NOTE: Some files based from MiNET.**

- [x] Navigators
- [x] Behaviors
  - [x] Behavior (Base)
  - [x] CreeperSwellBehavior
  - [x] EatBlockBehavior
  - [x] FindAttackableTargetBehavior
  - [x] FleeSunBehavior **NEW**
  - [x] FollowOwnerBehavior
  - [x] HorseEatBlockBehavior
  - [x] HurtByTargetBehavior
  - [x] JumpAttackBehavior
  - [x] LookAtPlayerBehavior
  - [x] MeleeAttackBehavior
  - [x] OwnerHurtByTargetBehavior
  - [x] OwnerHurtTargetBehavior
  - [x] PanicBehavior
  - [x] PathFinder
  - [x] RandomLookAroundBehavior
  - [x] SittingBehavior
  - [x] StrollBehavior
  - [x] TemptedBehavior
  - [x] WanderBehavior
- [ ] All Mobs
- [ ] Mob Spawner
- [ ] Implement Riding

### CODE REVIEW

**Comment by shdudwns** [comment]
The monsters do not move yet.
Only body and head move.

**Comment by Enes5519** [comment]
@shdudwns We knew that, but thanks for trying.

File: src/pocketmine/entity/behavior/navigator/TileNavigator.php [hunk 1]
```diff
@@ -70,10 +70,10 @@ public function navigate(Tile $from, Tile $to, int $maxAttempts = PHP_INT_MAX) :
 			$last = null;
 			
 			if(++$noOfAttempts > $maxAttempts){
-				$this->reConstructPath($path, $highScore);
+				return $this->reConstructPath($path, $highScore);
 			}
 			if($current->equals($to)){
-				$this->reConstructPath($path, $current);
+				return $this->reConstructPath($path, $current);
```
Comment (Enes5519) [inline src/pocketmine/entity/behavior/navigator/TileNavigator.php]: neden buraları return yaptın ?

**Comment by shdudwns** [comment]
Bug: Entities and flies are not hit, mobi ai is active when creative, mobi ai is not working when server is fired

**Comment by EmreTr1** [comment]
@shdudwns fixed

**Comment by shdudwns** [comment]
object, pocketmine\utils\MainLogger object, string /root/pmmp/, string /root/pmmp/plugins/)
^C10:20:55 CRITICAL > 레벨 "yungjun"의 틱을 처리하지 못했습니다: Call to a member function havePath() on null
10:20:55 CRITICAL > Error: "Call to a member function havePath() on null" (EXCEPTION) in "src/pocketmine/entity/behavior/WanderBehavior" at line 68

**Comment by shdudwns** [comment]
10:41:32 CRITICAL > 레벨 "yungjun"의 틱을 처리하지 못했습니다: Call to a member function nextFloat() on null
10:41:32 CRITICAL > Error: "Call to a member function nextFloat() on null" (EXCEPTION) in "src/pocketmine/entity/behavior/LookAtPlayerBehavior" at line 46

**Comment by Enes5519** [comment]
Yes

**Comment by shdudwns** [comment]
The error is still the same
@Enes5519

**Comment by shdudwns** [comment]
10:41:32 CRITICAL > 레벨 "yungjun"의 틱을 처리하지 못했습니다: Call to a member function nextFloat() on null
10:41:32 CRITICAL > Error: "Call to a member function nextFloat() on null" (EXCEPTION) in "src/pocketmine/entity/behavior/LookAtPlayerBehavior" at line 46

**Comment by shdudwns** [comment]
Entities and flies are not hit

**Comment by Enes5519** [comment]
@shdudwns This is not true. I just tried.

**Comment by Bruno99** [comment]
Guys, you just have to add the mobs? I want to praise you for the colossal work. You are the few who contribute more originality to mcpe. A special thanks to you. Continue in the same spirit!

**Comment by Bruno99** [comment]
Sorry for the trouble, but why did you decide to cover this request?

**Comment by Enes5519** [comment]
I don't like this mob ai. I code new

**Comment by EmreTr1** [comment]
i will continue with this branch

**Pull request closed without merging**